src/pam_hbac_config.c

```diff
@@ -281,13 +281,20 @@ ph_read_config(pam_handle_t *pamh,
 
     logger(pamh, LOG_DEBUG, "config file: %s", config_file);
 
+#ifdef HPUX
+    errno = 666;       // for some reason when invoked through pam in HPUX, fopen doesn't set errno on error
```

Comment:
It would be nice to use a constant from errno.h here. I think errno = EIO is a generic error that would suit here nicely. Another (and this is a pet peeve of mine, not as much technical...) is to use old C-style comments:
`/* or some reason when invoked through pam in HPUX, fopen doesn't set errno on error */`